test/JoinTest.cpp

```diff
@@ -469,3 +510,256 @@ TEST(JoinTest, invalidJoinVariable) {
+    auto expectedColumns = makeExpectedColumns(expectedVariables, expected);
+    auto join = Join{qec, leftTree, rightTree, 0, 0};
+    EXPECT_EQ(join.getDescriptor(), "Join on ?s");
+
+    qec->getQueryTreeCache().clearAll();
+    testJoinOperation(join, expectedColumns, true);
+    qec->getQueryTreeCache().clearAll();
+    testJoinOperation(join, expectedColumns, false);
+
+    auto joinSwitched = Join{qec, rightTree, leftTree, 0, 0};
+    qec->getQueryTreeCache().clearAll();
+    testJoinOperation(joinSwitched, expectedColumns, true);
+    qec->getQueryTreeCache().clearAll();
+    testJoinOperation(joinSwitched, expectedColumns, false);
+  };
+  auto U = Id::makeUndefined();
+  std::vector<IdTable> rightTables;
+  rightTables.push_back(makeIdTableFromVector({{U}}));
+  auto expected1 = makeIdTableFromVector({{U}});
+  performJoin(makeIdTableFromVector({{U}}), std::move(rightTables), expected1,
+              true);
+
+  rightTables.push_back(makeIdTableFromVector({{U}}));
+  auto expected2 = makeIdTableFromVector({{U}, {I(0)}});
+  performJoin(makeIdTableFromVector({{U}, {I(0)}}), std::move(rightTables),
+              expected2, true);
+
+  rightTables.push_back(makeIdTableFromVector({{I(0)}}));
+  auto expected3 = makeIdTableFromVector({{I(0)}, {I(0)}});
+  performJoin(makeIdTableFromVector({{U}, {I(0)}, {I(1)}}),
+              std::move(rightTables), expected3, false);
+
+  rightTables.push_back(makeIdTableFromVector({{U}, {I(0)}}));
+  auto expected4 = makeIdTableFromVector({{I(0)}, {I(0)}, {I(1)}});
+  performJoin(makeIdTableFromVector({{I(0)}, {I(1)}}), std::move(rightTables),
+              expected4, false);
+
+  rightTables.push_back(IdTable{1, ad_utility::makeUnlimitedAllocator<Id>()});
+  IdTable expected5{1, ad_utility::makeUnlimitedAllocator<Id>()};
+  performJoin(makeIdTableFromVector({{U}, {I(0)}, {I(1)}}),
+              std::move(rightTables), expected5, false);
+
+  rightTables.push_back(makeIdTableFromVector({{I(1)}}));
+  rightTables.push_back(makeIdTableFromVector({{I(2)}}));
+  auto expected6 = makeIdTableFromVector({{I(1)}});
+  performJoin(makeIdTableFromVector({{I(0)}, {I(1)}}), std::move(rightTables),
+              expected6, false);
+
+  rightTables.push_back(makeIdTableFromVector({{U}}));
+  rightTables.push_back(makeIdTableFromVector({{I(2)}}));
+  auto expected7 = createIdTableOfSizeWithValue(Join::CHUNK_SIZE, I(1));
+  performJoin(createIdTableOfSizeWithValue(Join::CHUNK_SIZE, I(1)),
+              std::move(rightTables), expected7, false);
+}
+
+// _____________________________________________________________________________
+TEST(JoinTest, errorInSeparateThreadIsPropagatedCorrectly) {
+  auto qec = ad_utility::testing::getQec();
+  RuntimeParameters().set<"lazy-index-scan-max-size-materialization">(0);
+  absl::Cleanup cleanup{[]() {
+    // Reset back to original value to not influence other tests.
+    RuntimeParameters().set<"lazy-index-scan-max-size-materialization">(
+        1'000'000);
+  }};
```

Comment:
Code duplication, see above.

Reply:
Maybe the `IndexHelpers` file would be a good fit, or maybe create a new `RuntimeParametersTestHelpers.h`.